packages/js/src/core/sign-in.ts

```diff
@@ -0,0 +1,50 @@
+import qs from 'query-string';
+
+import { withReservedScopes } from '../utils/scopes';
+
+const prompt = 'consent';
+const codeChallengeMethod = 'S256';
+
+enum ResponseType {
+  code = 'code',
+}
+
+export type SignInUriParameters = {
+  authorizationEndpoint: string;
+  clientId: string;
+  redirectUri: string;
+  codeChallenge: string;
+  state: string;
+  scopes?: string[];
+  resources?: string[];
+};
+
+export const generateSignInUri = ({
+  authorizationEndpoint,
+  clientId,
+  redirectUri,
+  codeChallenge,
+  state,
+  scopes,
+  resources,
+}: SignInUriParameters) => {
+  const queryStringWithoutResources = qs.stringify({
+    client_id: clientId,
+    redirect_uri: redirectUri,
+    code_challenge: codeChallenge,
+    code_challenge_method: codeChallengeMethod,
+    state,
+    response_type: ResponseType.code,
+    prompt,
+    scope: withReservedScopes(scopes),
+  });
+
+  const queryStringWithOnlyResources = (resources ?? []).reduce(
+    (accumulator, item) => `${accumulator}&resource=${item}`,
+    ''
+  );
+
+  const queryString = `${queryStringWithoutResources}${queryStringWithOnlyResources}`;
```

Comment:
why not just use qs.stringfy a list value?

Reply:
My bad